app/modules/scan/service.server.ts

```diff
@@ -138,3 +145,61 @@ export async function getScanByQrId({ qrId }: { qrId: string }) {
     });
   }
 }
+
+export async function createScanNote({ userId, qrId, latitude, longitude, manuallyGenerated }: {
+  userId?: string | null;
+  qrId: string;
+  latitude?: Scan["latitude"];
+  longitude?: Scan["longitude"];
+  manuallyGenerated?: boolean;
+}) {
+  try {
+    let message = "";
+    const { assetId, organizationId } = await getQr(qrId)
+    if (assetId) {
+      if (userId && userId != "anonymous") {
+        const user = await db.user.findFirst({
+          where: {
+            id: userId,
+          },
+          select: {
+            firstName: true,
+            lastName: true,
+          },
+        });
+
+        const userName = user?.firstName + " " + user?.lastName
```

Comment:
We had this issue before, i think if the user doesnt have first and last name it will print "undefined" as a string. Can you please double check it. Also you can see how we handle it in other notes.